lazy-pages/src/sys/unix.rs

```diff
@@ -29,24 +30,41 @@ thread_local! {
     static OLD_SIG_HANDLER: RefCell<SigHandler> = RefCell::new(SigHandler::SigDfl);
 }
 
-extern "C" fn handle_sigsegv(sig: i32, info: *mut siginfo_t, ucontext: *mut c_void) {
-    unsafe {
-        #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
-        let is_write = {
-            let ucontext = ucontext as *const nix::libc::ucontext_t;
+cfg_if! {
```

Comment:
Why can't we just use cfg attributes?
target os = linux and = macOS are incompatible on compile, so it should be fine

Reply:
They are not because `mcontext_t` have different definitions on these platforms

Also we have `#[cfg(all(target_os = "linux", target_arch = "x86_64"))]` and `#[cfg(all(target_os = "macos", target_arch = "aarch64"))]` and their negation will look like 
```
#[cfg(not(any(
    all(target_os = "linux", target_arch = "x86_64"), 
    all(target_os = "macos", target_arch = "aarch64")
)))]
```

Which is quite ugly

Reply:
Why do we need their negation?
We should write placeholder for macOS x86 and dont support any other platforms so (it will cause panic in compile-time)

Reply:
Ok, good suggestion. Will do